src/views/transaction-details/transaction-details.view.tsx

```diff
@@ -1,37 +1,110 @@
 import { ReactComponent as NewWindowIcon } from "src/assets/icons/new-window.svg";
 import { ReactComponent as SpinnerIcon } from "src/assets/icons/spinner.svg";
 import Typography from "src/views/shared/typography/typography.view";
-import { demoData } from "src/views/activity/demo-data";
-import { getTimeFromNow } from "src/utils/time";
 import Icon from "src/views/shared/icon/icon.view";
 import Chain from "src/views/transaction-details/components/chain/chain";
-import { getTransactionStatusText } from "src/domain";
-import { useEnvContext } from "src/contexts/env.context";
+import { getTransactionStatusText, Transaction } from "src/domain";
+import { useBridgeContext } from "src/contexts/bridge.context";
+import { useProvidersContext } from "src/contexts/providers.context";
+import { useUIContext } from "src/contexts/ui.context";
+import { parseError } from "src/adapters/error";
+import { AsyncTask, isMetamaskUserRejectedRequestError } from "src/utils/types";
 
 const TransactionDetails: FC = () => {
   const { transactionId } = useParams();
-  const env = useEnvContext();
-  const data = demoData.find((demo) => demo.id.toString() === transactionId);
-  const classes = useTransactionDetailsStyles({ status: data?.status || "initiated" });
+  const { openSnackbar } = useUIContext();
+  const { getTransactions, claim } = useBridgeContext();
+  const { account } = useProvidersContext();
+  const [transaction, setTransaction] = useState<AsyncTask<Transaction, string>>({
+    status: "pending",
+  });
+  const classes = useTransactionDetailsStyles({
+    status: transaction.status === "successful" ? transaction.data.status : undefined,
+  });
 
-  if (!data) {
+  const onClaim = () => {
+    if (transaction.status === "successful" && transaction.data.status === "on-hold") {
+      const tx = transaction.data;
+      void claim({
+        originalTokenAddress: tx.token.address,
+        amount: tx.amount,
+        originalNetwork: tx.originNetwork.networkId.toString(),
+        destinationNetwork: tx.destinationNetwork,
+        destinationAddress: tx.destinationAddress,
+        index: tx.depositCount,
+        smtProof: tx.merkleProof,
+        globalExitRootNum: tx.exitRootNumber,
+        mainnetExitRoot: tx.mainExitRoot,
+        rollupExitRoot: tx.rollupExitRoot,
+      }).catch((error) => {
+        if (isMetamaskUserRejectedRequestError(error) === false) {
+          void parseError(error).then((parsed) => {
+            openSnackbar({
+              type: "error",
+              parsed,
+            });
+          });
+        }
+      });
+    }
+  };
+
+  useEffect(() => {
+    if (account.status === "successful") {
+      // ToDo: Get all the data only for the right bridge
+      void getTransactions({ ethereumAddress: account.data })
+        .then((transactions) => {
+          const foundTransaction = transactions.find((tx) => {
+            const id = `${tx.destinationNetwork.networkId}-${tx.depositCount}`;
```

Comment:
I see that you use a lot here and there:

```
const id = `${tx.destinationNetwork.networkId}-${tx.depositCount}`;
```

Why not adding it to the `Transaction` domain when calling `getTransactions`? At least doing this we'll have the `id` centralized in one point. It's safer If we need to change it in the future as well.

Reply:
Yeah much better. Done.